### PR TITLE
Aymeric/speedup incremental build

### PR DESCRIPTION
**Pull request by achataigner**
Apply required changes from https://github.com/Slicer/Slicer/pull/7001

### CODE REVIEW

**Comment by cpinter** [comment]
Looks good to me, thank you! @lassoan @Sunderlandkyl any objections?

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,14 @@ set(EXTENSION_BUILD_SUBDIRECTORY inner-build)
 
 set(SUPERBUILD_TOPLEVEL_PROJECT inner)
 
+# Prevent unnecessary re-linking on Linux
+#
+# See https://discourse.slicer.org/t/build-when-git-commit-is-changed/29607/12
```
Comment (lassoan) [inline CMakeLists.txt:28]: Source code comments should be sufficiently self-contained. It is nice to provide some web URLs for details, but we should not rely on that. So, please add a bit more details here (similarly how it is done in Slicer core).

File: DicomRtImportExport/ConversionRules/vtkPlanarContourToClosedSurfaceConversionRule.h [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #define __vtkPlanarContourToClosedSurfaceConversionRule_h
 
 // Slicer include
-#include <vtkSlicerVersionConfigure.h>
+#include <vtkSlicerVersionConfigureMinimal.h>
```
Comment (lassoan) [inline DicomRtImportExport/ConversionRules/vtkPlanarContourToClosedSurfaceConversionRule.h]: `vtkSlicerVersionConfigureMinimal.h` is not available in current Slicer stable branch. The right file should be included depending on Slicer version; or this change should be delayed until a new stable version is released (probably in a couple of months).

File: DicomRtImportExport/ConversionRules/vtkPlanarContourToRibbonModelConversionRule.h [hunk 3]
```diff
@@ -22,7 +22,7 @@
 #define __vtkPlanarContourToRibbonModelConversionRule_h
 
 // Slicer include
-#include <vtkSlicerVersionConfigure.h>
+#include <vtkSlicerVersionConfigureMinimal.h>
```
Comment (lassoan) [inline DicomRtImportExport/ConversionRules/vtkPlanarContourToRibbonModelConversionRule.h]: same as above

File: DoseVolumeHistogram/Logic/vtkSlicerDoseVolumeHistogramModuleLogic.cxx [hunk 4]
```diff
@@ -78,7 +78,7 @@
 #include <set>
 
 // Slicer includes
-#include <vtkSlicerVersionConfigure.h>
+#include <vtkSlicerVersionConfigureMinimal.h>
```
Comment (lassoan) [inline DoseVolumeHistogram/Logic/vtkSlicerDoseVolumeHistogramModuleLogic.cxx]: same as above

File: ExternalBeamPlanning/Widgets/qSlicerMockDoseEngine.cxx [hunk 5]
```diff
@@ -39,7 +39,7 @@
 #include <vtkPolyData.h>
 
 // Slicer includes
-#include <vtkSlicerVersionConfigure.h>
+#include <vtkSlicerVersionConfigureMinimal.h>
```
Comment (lassoan) [inline ExternalBeamPlanning/Widgets/qSlicerMockDoseEngine.cxx]: same as above

**Review by lassoan — COMMENTED** [review]
Thanks for working on this. I've added a few comments.

File: DicomRtImportExport/ConversionRules/vtkPlanarContourToClosedSurfaceConversionRule.h (hunk 2 above)
Comment (cpinter) [reply DicomRtImportExport/ConversionRules/vtkPlanarContourToClosedSurfaceConversionRule.h]: Ah, good point. We don't fix the SlicerRT hash anymore for the stable version.
https://github.com/Slicer/ExtensionsIndex/blob/5.2/SlicerRT.s4ext#L10

Comment (achataigner) [reply DicomRtImportExport/ConversionRules/vtkPlanarContourToClosedSurfaceConversionRule.h]: OK, removed

File: CMakeLists.txt (hunk 1 above)
Comment (achataigner) [reply CMakeLists.txt:28]: done

**Comment by achataigner** [comment]
Is it ok to merge ?

**Comment by achataigner** [comment]
Anything to add ? Is it ready to be merged ?

**Review by lassoan — APPROVED** [review]
It looks good to me. @cpinter if you don't have any comments then we can merge it.

**Comment by achataigner** [comment]
@cpinter Is it ok ?

**Comment by achataigner** [comment]
@cpinter Do you know when you would have time to merge this branch ?

**Comment by lassoan** [comment]
I think it is safe to merge this, so I do it now. If @cpinter had any concerns then it can be reverted later.

**Comment by achataigner** [comment]
Thank you Csaba for the fix about help text

**Pull request merged**